idaes/models/properties/general_helmholtz/components/co2.py

```diff
@@ -0,0 +1,101 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""This module provides co2 property expressions
+
+Vesovic, V., W.A. Wakeham, G.A. Olchowy, J.V. Sengers, J.T.R. Watson, J.
+   Millat, (1990). "The transport properties of carbon dioxide." J. Phys.
+   Chem. Ref. Data, 19, 763-808.
+Fenghour, A., W.A. Wakeham, V. Vesovic, (1998). "The Viscosity of Carbon
+   Dioxide." J. Phys. Chem. Ref. Data, 27, 31-44.
+"""
+
+import pyomo.environ as pyo
+
+
+def _thermal_conductivity(blk, delta, tau, on_blk=None):
```

Comment:
Yes, something like cool prop, but we'd have to figure out all the possible forms.  There are probably a limited number of possibilities, but it's not trivial.

Reply:
There are ways to parse a string representation of an expression into a Pyomo `expression` - we do this in the ALAMOpy wrapper. Seeing as we have a standard set of variable names that would be involved, this is probably not too hard to do, and then the JSON/YAML file could just contain a string representation of the form.

Reply:
Yeah, but that's just a partial solution.  Then you have the whole EOS.

Reply:
I thought there was a standard for for Helmholtz EoS (albeit with a lot of parameters that aren't needed in all cases)?

Reply:
Yeah.  I think there is a standard for the EOS.  There is a really complex function and in practice you end up with a simpler subset of terms, so they don't all look exactly the same.  Reading the parameters from a json file is totally something I can do.  It could be a while before we get there though depending on how priorities shake out.  It's less easy on the C++ side than Python.

Reply:
This is definitely something for the future - just wanted to make sure we discussed it a little here.